random_numbers/include/random_numbers.inc

```diff
@@ -16,123 +16,37 @@
-  implicit none
-  private
-
-  !> @brief Type to hold a stream of randomly generated numbers
-  !> @ingroup random_numbers_mod
-  type randomNumberStream
-    type(randomNumberSequence) :: theNumbers
-  end type randomNumberStream
-
-  !> Returns scalar, 1 or 2 D random real numbers
-  !!
-  !> @param stream @ref randomNumberStream to generate from
-  !> @param[out] number output number(s)
-  !> @ingroup random_numbers_mod
-  interface getRandomNumbers
-    module procedure getRandomNumber_Scalar, getRandomNumber_1D, getRandomNumber_2D
-  end interface getRandomNumbers
-
-  !> Initializes stream for generating random numbers.
-  !> @ingroup random_numbers_mod
-  interface initializeRandomNumberStream
-    module procedure initializeRandomNumberStream_S, initializeRandomNumberStream_V
-  end interface initializeRandomNumberStream
-
-  public :: randomNumberStream,                             &
-            initializeRandomNumberStream, getRandomNumbers, &
-            constructSeed
-!> @addtogroup random_numbers_mod
-!> @{
-contains
-  ! ---------------------------------------------------------
-  !> Initialization
-  function initializeRandomNumberStream_S(seed) result(new)
-    integer, intent( in)     :: seed
-    type(randomNumberStream) :: new
-
-    new%theNumbers = new_RandomNumberSequence(seed)
-
-  end function initializeRandomNumberStream_S
-  ! ---------------------------------------------------------
-  function initializeRandomNumberStream_V(seed) result(new)
-    integer, dimension(:), intent( in) :: seed
-    type(randomNumberStream)           :: new
-
-    new%theNumbers = new_RandomNumberSequence(seed)
-
-  end function initializeRandomNumberStream_V
-  ! ---------------------------------------------------------
   !> Draws random scalar
-  subroutine getRandomNumber_Scalar(stream, number)
+  subroutine GET_RANDOM_NUMBER_0D_(stream, number)
     type(randomNumberStream), intent(inout) :: stream
-    real,                     intent(  out) :: number
+    real(FMS_RN_KIND_),       intent(  out) :: number
+
+    number = real(getRandomReal(stream%theNumbers), FMS_RN_KIND_)
+  end subroutine
 
-    number = real(getRandomReal(stream%theNumbers))
-  end subroutine getRandomNumber_Scalar
-  ! ---------------------------------------------------------
   !> Draws random 1D array
-  subroutine getRandomNumber_1D(stream, numbers)
-    type(randomNumberStream), intent(inout) :: stream
-    real, dimension(:),       intent(  out) :: numbers
+  subroutine GET_RANDOM_NUMBER_1D_(stream, numbers)
+    type(randomNumberStream),         intent(inout) :: stream
+    real(FMS_RN_KIND_), dimension(:), intent(  out) :: numbers
 
     ! Local variables
     integer :: i
 
     do i = 1, size(numbers)
-      numbers(i) = real(getRandomReal(stream%theNumbers))
+      call GET_RANDOM_NUMBER_0D_(stream, numbers(i))
     end do
-  end subroutine getRandomNumber_1D
-  ! ---------------------------------------------------------
+  end subroutine
```

Comment:
One can randomly distribute random numbers in a linear chain of pockets which may create more randomness for multidimensional arrays (2d, 3d, 4d, ... etc.) down the road .... just a thought !